### PR TITLE
[Feature](IP) support ipv4/ipv6 with inverted index and conjuncts for query

### PR DESCRIPTION
**Pull request by amorynan**
## Proposed changes
support data type ipv4/ipv6 with inverted index 
and then we can query like "> or < or >= or <=  or in/not in " this conjuncts expr for ip with inverted index speeding up
Issue Number: close #xxx

### CODE REVIEW

**Review by morrySnow — DISMISSED** [review]

File: be/src/olap/rowset/segment_v2/column_reader.cpp [hunk 1]
```diff
@@ -573,7 +573,7 @@ Status ColumnReader::_load_inverted_index_index(
                         "create StringTypeInvertedIndexReader error: {}", e.what());
             }
         }
-    } else if (is_numeric_type(type)) {
+    } else if (is_numeric_type(type) || is_ip_type(type)) {
```
Comment (xiaokang) [inline be/src/olap/rowset/segment_v2/column_reader.cpp:576]: is_numeric_type() should be the same as field_is_numeric_type() and is_ip_type() can be removed here.

File: be/src/olap/utils.h [hunk 2]
```diff
@@ -260,6 +260,10 @@ constexpr bool is_numeric_type(const FieldType& field_type) {
            field_type == FieldType::OLAP_FIELD_TYPE_BOOL;
 }
 
+constexpr bool is_ip_type(const FieldType& field_type) {
```
Comment (xiaokang) [inline be/src/olap/utils.h:263]: not necessary

File: be/src/olap/rowset/segment_v2/inverted_index_writer.cpp [hunk 3]
```diff
@@ -695,6 +695,8 @@ Status InvertedIndexColumnWriter::create(const Field* field,
         M(FieldType::OLAP_FIELD_TYPE_DECIMAL128I)
         M(FieldType::OLAP_FIELD_TYPE_DECIMAL256)
         M(FieldType::OLAP_FIELD_TYPE_BOOL)
+        M(FieldType::OLAP_FIELD_TYPE_IPV4)
+        M(FieldType::OLAP_FIELD_TYPE_IPV6)
```
Comment (xiaokang) [inline be/src/olap/rowset/segment_v2/inverted_index_writer.cpp:699]: Where is the code for transform encoding from little endian to big endian?

Comment (airborne12) [reply be/src/olap/rowset/segment_v2/inverted_index_writer.cpp:699]: in key_coder's full_encoding function.

Comment (amorynan) [reply be/src/olap/rowset/segment_v2/inverted_index_writer.cpp:699]: writer & reader part both has key_coder's full_encoding function before add_doc and seek in bkd

File: be/src/olap/rowset/segment_v2/column_reader.cpp (hunk 1 above)
Comment (amorynan) [reply be/src/olap/rowset/segment_v2/column_reader.cpp:576]: ok I will make the same

**Comment by amorynan** [comment]
run buildall

File: be/src/olap/rowset/segment_v2/inverted_index_writer.cpp (hunk 3 above)
Comment (xiaokang) [reply be/src/olap/rowset/segment_v2/inverted_index_writer.cpp:699]: ok

**Review by xiaokang — APPROVED** [review]
LGTM

**Review by airborne12 — APPROVED** [review]
LGTM

**Pull request merged**